archive/walk.go

```diff
@@ -245,6 +251,100 @@ func (c Chunk) Range() string {
 	return fmt.Sprintf("[%d-%d]", c.First, c.Last)
 }
 
+type chunkReader struct {
+	io.Reader
+	io.Closer
+	totalSize int64
+	readSize  int64
+}
+
+// newChunkReader returns an io.ReadCloser for this chunk. If the chunk has a seek
+// index and if the provided span skips part of the chunk, the seek index will
+// be used to limit the reading window of the chunk's reader.
+func newChunkReader(ctx context.Context, chunk Chunk, ark *Archive, span nano.Span) (*chunkReader, error) {
+	cspan := chunk.Span()
+	span = cspan.Intersect(span)
+	if span.Dur == 0 {
+		return nil, errors.New("chunk span does intersect provided span")
+	}
+	r, err := iosrc.NewReader(ctx, chunk.Path(ark))
+	if err != nil {
+		return nil, err
+	}
+	cr := &chunkReader{
+		Reader:    r,
+		Closer:    r,
+		totalSize: chunk.Size,
+		readSize:  chunk.Size,
+	}
+	if span == cspan {
+		return cr, nil
+	}
+	finder, err := microindex.NewFinder(ctx, resolver.NewContext(), chunk.seekIndexPath(ark))
+	if zqe.IsNotFound(err) {
+		return cr, nil
+	} else if err != nil {
+		return nil, err
+	}
```

Comment:
Might be more readable like this.
```suggestion
	if err != nil {
		if zqe.IsNotFound(err) {
			return cr, nil
		}
		return nil, err
	}
```